README.md

```diff
@@ -50,3 +50,28 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+
+///
+
+The swap function performs an array flip operation. When the input is the number of cities n, the time complexity is O(n)
+
+The calculateroutelength function needs to traverse the entire path and calculate the distance, and the time complexity is O(n)
+
+Since the maximum execution stop is set to 1000 times, the number of loops is O(n)
+
+The total time complexity is theta(n)
+
+If the number of iterations is not limited, for n cities, the worst case number of iterations is O(n!)
```

Comment:
Not following you here. If the number of iterations is not limited, how would it ever terminate?